db/sovereign-db/src/ledger_db/rpc.rs

```diff
@@ -78,19 +84,67 @@ impl LedgerRpcProvider for LedgerDB {
         //      and use an iterator instead of querying for each slot individually
-        let mut out = Vec::with_capacity(slot_ids.len());
-        for slot_id in slot_ids {
-            let slot_num = self.resolve_slot_identifier(slot_id)?;
-            out.push(match slot_num {
-                Some(num) => {
-                    if let Some(stored_slot) = self.db.get::<SlotByNumber>(&num)? {
-                        Some(self.populate_slot_response(num.into(), stored_slot, query_mode)?)
-                    } else {
-                        None
+
+        // First resolve the slot identifiers and store the slot numbers
+        let slot_nums_result: Result<Vec<(usize, Option<SlotNumber>)>, anyhow::Error> = slot_ids
+            .iter()
+            .enumerate()
+            .map(
+                |(idx, slot_id)| match self.resolve_slot_identifier(slot_id) {
+                    Ok(slot_number) => Ok((idx, slot_number)),
+                    Err(e) => Err(e),
+                },
+            )
+            .collect();
+
+        let mut slot_nums = match slot_nums_result {
+            Ok(slot_numbers) => slot_numbers,
+            Err(e) => return Err(e),
+        };
+
+        // Then sort the slot numbers by requested order
+        slot_nums.sort_unstable_by(|(_, slot_id_a), (_, slot_id_b)| slot_id_a.cmp(slot_id_b));
+
+        // Then identify sequential ranges
+        let mut slot_ranges: Vec<QueriedRange> = Vec::new();
+        let mut current_range: Option<QueriedRange> = None;
+
+        for (orig_idx, slot_num_opt) in slot_nums {
+            if let Some(num) = slot_num_opt {
+                match current_range {
+                    Some(range) if SlotNumber(Into::<u64>::into(range.inner.end) + 1) == num => {
+                        current_range = Some(QueriedRange {
+                            inner: range.inner.start..num,
+                            original_indices: range.original_indices,
+                        });
+                    }
+                    _ => {
+                        if let Some(range) = current_range {
+                            slot_ranges.push(range);
+                        }
+                        current_range = Some(QueriedRange {
+                            inner: num..SlotNumber(Into::<u64>::into(num) + 1),
+                            original_indices: vec![orig_idx],
+                        });
                     }
                 }
-                None => None,
-            })
+            }
+        }
+
+        // For each identified range, make a single query to fetch tx's
+        // From the corresponding blocks, then store into result vector.
+        let mut out = Vec::with_capacity(slot_ids.len());
+        for slot_range in slot_ranges {
+            let stored_slots = self.get_slot_range(&slot_range.inner)?;
+
+            for ((queried_slot, original_idx), slot_num) in stored_slots
+                .iter()
```

Comment:
The build is failing because of this borrow. Looks like it needs to change to `into_iter()`. 

On a related note - looks like CI has failed a couple of times in a row. Is your repo out of sync with the upstream? If not, you might want to try using the Makefile we've included! From the root directory (`sovereign/`), just run `make check` - it should catch pretty much everything that might cause CI to fail.

Reply:
Was able to run this after rebasing - I made a slight error in forgetting one of the steps that made things fail, sorry about that. Most of the checks are passing now.